gradle/compile.gradle

```diff
@@ -36,9 +37,15 @@ tasks.withType(JavaCompile) {
     // while compiling.
     options.compilerArgs = ['-Xlint:-options', '-Xlint:-path']
     options.incremental = true
+    inputs.property('javaVendor') {
+        java7Home ? 'Oracle Corporation' : javaVendor
```

Comment:
:question: I guess you do this because you don't know what the target JDK is? What about using our internal `org.gradle.jvm.toolchain.internal.JavaInstallationProbe`? If you look at `org.gradle.jvm.toolchain.internal.JavaInstallationProbeTest` you can see that it can give you a name for a Java installation by pointing at its install path (results are cached), and it would handle more JDKs.